### PR TITLE
R21C HISTORY output specs (precip, land, landice, ...)

### PR DESCRIPTION
**Pull request by gmao-rreichle**
This PR proposes updated R21C HISTORY specs on top of @sdrabenh's R21C branch.
See also [matching PR #637 in GEOSgcm_GridComp](https://github.com/GEOS-ESM/GEOSgcm_GridComp/pull/637). 
These updates will eventually need to find their way into the default and/or FP file specs in the 'develop' branch, which will require another PR.  

----
**Details:**

Updates are focused on (but not entirely limited to) land and landice outputs.

For precip outputs, clarified variable names for rain vs. snow and model vs. corrected.

For land outputs in the "lnd" collection, appended/inserted "LAND" in the nc4 variable name wherever there's any chance at all that users could be confused that the output is only over land tiles.  (The nc4 variable names for soil moisture and soil temperature, which are probably the most commonly used outputs, ,were not changed.) 

Updates were applied to both the lat/lon and cube-sphere collection definitions.  (For "lnd", updates are also applied to the forecast collection, which is not written out for R21C.  This should help with getting the updates into 'develop' later.) 

Removed obsolete SWLAND output from "lfo" collections.

Made variables that are exported in R21C consistent across land and landice (incl. runoff).

Fixed sign issue for GHTSKIN in landice.

Added interception water storage (CAPAC) to lnd output (suggested by Mike B based on user feedback)

Still to do:

1. There seem to be some funky choices in the list of outputs for the ocn collections.

2. Add runoff (and other variables?) averaged across land and landice.  [**DONE** @lcandre2, @gmao-rreichle 16Nov22]
3. TBD: Is there a difference for DRPAR & DFPAR output from SOLAR vs SURFACE? --> only roundoff [**DONE** @lcandre2 16Nov22]
4. TBD: Is there a difference for ALBEDO output from SOLAR vs SURFACE?  --> only roundoff [**DONE** @lcandre2 16Nov22]
5. TBD: Community request for output of nadir albedo (in addition to broadband albedo).  [**DONE** @lcandre2 16Nov22]
6. Resolve conflicting signs for GHTSKIN (GHLAND?) in landice and land collections.   [**DONE** @lcandre2 16Nov22]
7. The "vertically integrated" ("int") collection is only defined for lat/lon.  Is the omission of the cube-sphere collection intentional? 
8. **For "lnd", when merging with 'develop', make sure not to inadvertently drop the new "PEATCLSM" output variables.** 
9. **Add land surface constants output** as in MERRA-2: const_2d_lnd_Nx (M2CONXLND) "Constant Land‐Surface Parameters".  I think in MERRA-2, this was done manually (Rob and Qing working with matlab).  It would be preferable to have this formalized through HISTORY, if at all possible.  Needs thought.

cc: @elakkraoui @lcandre2 @gmao-qliu @lltakacs @sdrabenh @rcullath @rdkoster @rlucches 
----
From @lcandre2 (11/18/22):  
I merged my branch /feature/lcandre2/r21c_landice_updates into /feature/rreichle/r21c_updates  and made some modifications based on a meeting with @gmao-rreichle on 16Nov22. 

There are outstanding issues that require input from others to resolve. See notes in files changed.

### CODE REVIEW

**Comment by rcullath** [comment]
Ok, thanks Rolf.

~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~
Richard Cullather
ESSIC, University of Maryland at College Park
c/o NASA/GSFC Code 610.1 (Bldg. 33 / Rm. G114)     Tel: +1 301 614-5668
8800 Greenbelt Road                                 Fax: +1 301 614-6246
Greenbelt, MD 20771                   e-mail: ***@***.***

From: Rolf Reichle ***@***.***>
Sent: Friday, September 09, 2022 6:06 PM
To: GEOS-ESM/GEOSgcm_App ***@***.***>
Cc: Cullather, Richard I. (GSFC-610.1)[UNIVERSITY OF MARYLAND] ***@***.***>; Assign ***@***.***>
Subject: [EXTERNAL] Re: [GEOS-ESM/GEOSgcm_App] R21C HISTORY output specs (precip, land, landice, ...) (PR #356)


Assigned #356<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSgcm_App%2Fpull%2F356&data=05%7C01%7Crichard.cullather%40nasa.gov%7C4883298911394b0696dc08da92af8d10%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983579925703285%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=oMAY%2FhQiAe4njpq1ipXs%2BrNH%2BtOnMwKmBj8%2F0xa9Qnc%3D&reserved=0> to @rcullath<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frcullath&data=05%7C01%7Crichard.cullather%40nasa.gov%7C4883298911394b0696dc08da92af8d10%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983579925703285%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=RZKa9ULrVsPBzuyRMImknUjNcNdY%2Bx9MpQOqNF7pTpE%3D&reserved=0>.

-
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSgcm_App%2Fpull%2F356%23event-7359674878&data=05%7C01%7Crichard.cullather%40nasa.gov%7C4883298911394b0696dc08da92af8d10%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983579925703285%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=%2BpVT%2FjNgUkL2OzvkFMXZEylyBhUa17oGdDvs6BrzeV8%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAPJB4W42B6NWS5PCDTX37PTV5OYGHANCNFSM6AAAAAAQJBI7OM&data=05%7C01%7Crichard.cullather%40nasa.gov%7C4883298911394b0696dc08da92af8d10%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983579925703285%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=IWR6GxfyvH4t%2Bd0ORjRWveZH%2Br1EIE%2F1OwgWJOKYQQQ%3D&reserved=0>.
You are receiving this because you were assigned.Message ID: ***@***.******@***.***>>

**Comment by rcullath** [comment]
Thanks for the update.

~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~~~~ ~~
Richard Cullather
ESSIC, University of Maryland at College Park
c/o NASA/GSFC Code 610.1 (Bldg. 33 / Rm. G114)     Tel: +1 301 614-5668
8800 Greenbelt Road                                 Fax: +1 301 614-6246
Greenbelt, MD 20771                   e-mail: ***@***.***

From: Rolf Reichle ***@***.***>
Sent: Friday, September 09, 2022 6:09 PM
To: GEOS-ESM/GEOSgcm_App ***@***.***>
Cc: Cullather, Richard I. (GSFC-610.1)[UNIVERSITY OF MARYLAND] ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] [GEOS-ESM/GEOSgcm_App] R21C HISTORY output specs (precip, land, landice, ...) (PR #356)


This PR proposes updated R21C HISTORY specs on top of @sdrabenh<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fsdrabenh&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=oeZDpopjTtDJMBFKMQYCGith8x62hDsj5Qlk%2B%2Bc4iA4%3D&reserved=0>'s R21C branch.
A matching PR will be created for the GEOSgcm_GridComp repository.
These updates will eventually need to find their way into the default and/or FP file specs in the 'develop' branch, which will require another PR.

________________________________

Details:

Updates are focused on (but not entirely limited to) land and landice outputs.

For precip outputs, clarified variable names for rain vs. snow and model vs. corrected.

For land outputs in the "lnd" collection, appended/inserted "LAND" in the nc4 variable name wherever there's any chance at all that users could be confused that the output is only over land tiles. (The nc4 variable names for soil moisture and soil temperature, which are probably the most commonly used outputs, ,were not changed.)

Updates were applied to both the lat/lon and cube-sphere collection definitions. (For "lnd", updates are also applied to the forecast collection, which is not written out for R21C. This should help with getting the updates into 'develop' later.)

Removed obsolete SWLAND output from "lfo" collections.

Still to do:

  1.  There seem to be some funky choices in the list of outputs for the ocn collections.
  2.  Add runoff (and other variables?) averaged across land and landice.
  3.  TBD: Is there a difference for DRPAR & DFPAR output from SOLAR vs SURFACE?
  4.  TBD: Is there a difference for ALBEDO output from SOLAR vs SURFACE?
  5.  TBD: Community request for output of nadir albedo (in addition to broadband albedo).
  6.  Resolve conflicting signs for GHTSKIN (GHLAND?) in landice and land collections.

I'll keep working on this, but please feel free take a look and add your 2c.

cc: @elakkraoui<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Felakkraoui&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=ZrKMn8SB5QhqW%2BJMvSFIRX83RtEDpKFDW%2FmgHeSGcUg%3D&reserved=0> @lcandre2<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Flcandre2&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=7Y108uZDTRvqQPW0fl%2BBPdUn4nMnUTVn0IflassDx9U%3D&reserved=0> @gmao-qliu<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgmao-qliu&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=hfcgX0JJbLlDY0DlOhByL03%2BBt27yv9930n%2FHejvR1k%3D&reserved=0> @lltakacs<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Flltakacs&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=96DTfaApA3LOGxayr%2FetJ6DSKHW7A7Rzp4Vg2XN1TfA%3D&reserved=0> @sdrabenh<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fsdrabenh&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=oeZDpopjTtDJMBFKMQYCGith8x62hDsj5Qlk%2B%2Bc4iA4%3D&reserved=0> @rcullath<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frcullath&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Ai%2FkLUCa76B67q5icbgfNYcEhXbmchohYXw5d1GLstE%3D&reserved=0>

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/GEOS-ESM/GEOSgcm_App/pull/356<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSgcm_App%2Fpull%2F356&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=FN2P%2BG1ntByrO%2FwRIcaJiLBIQHROetNbXgMUklYRt1Q%3D&reserved=0>

Commit Summary

  *   2a518c8<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSgcm_App%2Fpull%2F356%2Fcommits%2F2a518c8b3470fc421ff831f4eb22444e65a449b1&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=0PKOz2G9w4di%2BNdNIvXMyOSGy0UeY3fOeCAJfmqL2Yc%3D&reserved=0> revised R21C HISTORY output specs for precip and land (to be continued)

File Changes

(1 file<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSgcm_App%2Fpull%2F356%2Ffiles&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=UthQZrpR7cqB6Qfxz4W4yBCnKPaoOLhM8I52WcT7Bzk%3D&reserved=0>)

  *   M HISTORY_R21C.rc.tmpl<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSgcm_App%2Fpull%2F356%2Ffiles%23diff-3bc9f9ca164fb1df73cf37f615d6d9c95bf8dbd6c8c0c09cbb9e99596d005ae3&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229107675%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=nG%2B1946BWkoqXC3A1p7KEN%2BDlL%2Frq4If0y5m2R0EViA%3D&reserved=0> (258)

Patch Links:

  *   https://github.com/GEOS-ESM/GEOSgcm_App/pull/356.patch<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSgcm_App%2Fpull%2F356.patch&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229263899%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=O3JcAG%2BfHkVlm7G%2Fps6zSmV71fIdEHARTkfKxIDgPX4%3D&reserved=0>
  *   https://github.com/GEOS-ESM/GEOSgcm_App/pull/356.diff<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSgcm_App%2Fpull%2F356.diff&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229263899%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=68JxUFaX%2BgjEktCoo52X6hIO330%2FDjHurLu5oYBBcRs%3D&reserved=0>

-
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSgcm_App%2Fpull%2F356&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229263899%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=CMSnmWzddLjy0p4qQnFI3dzHKJvB5DgJxMOOX5gXDN8%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAPJB4W3TXORYUUQNXEP4E73V5OYOLANCNFSM6AAAAAAQJBI7OM&data=05%7C01%7Crichard.cullather%40nasa.gov%7C745a8b34f7f04635b5f108da92afd9eb%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637983581229263899%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=usG0DOnEdkUiTGmM6MadepqyTdJRY2H9nAEUZUyD38k%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

**Comment by sanAkel** [comment]
@elakkraoui and _others_,
here is a suggested addition that writes out `2d` fields for the ocean surface.

[ocean_forcing.rc.txt](https://github.com/GEOS-ESM/GEOSgcm_App/files/9569850/ocean_forcing.rc.txt)

**Comment by lcandre2** [comment]
> @elakkraoui and _others_, here is a suggested addition that writes out `2d` fields for the ocean surface.
> 
> [ocean_forcing.rc.txt](https://github.com/GEOS-ESM/GEOSgcm_App/files/9569850/ocean_forcing.rc.txt)

@sanAkel, here you have RUNSURF for runoff. This does not include glacier runoff. Do you want me to change this?

File: HISTORY_R21C.rc.tmpl [hunk 1]
```diff
@@ -1032,7 +1033,7 @@ COLLECTIONS: 'asm_inst_3hr_glo_L@HIST_IMx@HIST_JM_p48'
   flx_tavg_1hr_glo_C@AGCM_IMx@AGCM_IMx6_slv.end_date:     >>>IOEDATE<<< ,
   flx_tavg_1hr_glo_C@AGCM_IMx@AGCM_IMx6_slv.end_time:     >>>IOETIME<<< ,
   flx_tavg_1hr_glo_C@AGCM_IMx@AGCM_IMx6_slv.fields:      'LHFX'           , 'SURFACE'     , 'EFLUX'      ,
-                                                    'EVAP'           , 'SURFACE'     ,
+                                                    'EVAP'           , 'SURFACE'     ,                      ##
```
Comment (lcandre2) [inline HISTORY_R21C.rc.tmpl:1036]: **@lcandre2 is determining if EVAP=EVAPOUT
ISSUE:
EVAP is both an import and export of SURFACE
flx collection:  EVAP from surface 
ocn collection:  EVAPOUT from surface
lnd collection: separate EVAP components
glc collection: EVAPOUT from landice

File: HISTORY_R21C.rc.tmpl [hunk 2]
```diff
@@ -1525,7 +1531,7 @@ COLLECTIONS: 'asm_inst_3hr_glo_L@HIST_IMx@HIST_JM_p48'
                                                     'FLNS'         , 'IRRAD'    , 'LWGNT'   ,
                                                     'OLR'          , 'IRRAD'    , 'LWTUP'   ,
                                                     'RSRS'         , 'SOLAR'    , 'SWGNT'   ,
-                                                    'RUNSURF'      , 'SURFACE'  , 'RUNOFF'  ,
+                                                    'RUNSURF'      , 'SURFACE'  , 'RUNOFF'  , #is this the right runoff?
```
Comment (lcandre2) [inline HISTORY_R21C.rc.tmpl:1534]: @sanAkel I think that this should actually be:
'RUNOFF', 'SURFACE', 'RUNOFFTOT',
in order to include glacier runoff

File: HISTORY_R21C.rc.tmpl [hunk 3]
```diff
@@ -1537,8 +1543,8 @@ COLLECTIONS: 'asm_inst_3hr_glo_L@HIST_IMx@HIST_JM_p48'
                                                     'DFNIR'        , 'SOLAR'    , 'NIRDF'   ,
                                                     'DRNIR'        , 'SOLAR'    , 'NIRDR'   ,
                                                     'TAUX;TAUY'    , 'SURFACE'  ,
-                                                    'SHWTR'        , 'SURFACE'  , 'HFLUXWTR',
-                                                    'HLATWTR'      , 'SURFACE'  , 'EFLUXWTR',
+                                                    'SHWTR'        , 'SURFACE'  , 'SHOCN',
```
Comment (lcandre2) [inline HISTORY_R21C.rc.tmpl:1546]: @sanAkel @elakkraoui I changed the variable name to be consistent with land and landice.

File: HISTORY_R21C.rc.tmpl [hunk 4]
```diff
@@ -3993,27 +3998,74 @@ COLLECTIONS: 'asm_inst_3hr_glo_L@HIST_IMx@HIST_JM_p48'
...
-  glc_tavg_1hr_glo_L360x181_slv.regrid_name:            'PE@AGCM_IMx@AGCM_JM-CF' ,
-  glc_tavg_1hr_glo_L360x181_slv.regrid_exch:            'tile.bin',
   glc_tavg_1hr_glo_L@HIST_IMx@HIST_JM_slv.grid_label:    PC@HIST_IMx@HIST_JM-DC ,
   glc_tavg_1hr_glo_L@HIST_IMx@HIST_JM_slv.end_date:      >>>IOEDATE<<< ,
   glc_tavg_1hr_glo_L@HIST_IMx@HIST_JM_slv.end_time:      >>>IOETIME<<< ,
-  glc_tavg_1hr_glo_L@HIST_IMx@HIST_JM_slv.fields:       'SNOWDP_GL'    , 'LANDICE'  ,
-                                                    'SNOMAS_GL'    , 'LANDICE'  ,
-                                                    'ASNOW_GL'     , 'LANDICE'  ,
-                                                    'RUNOFF'       , 'LANDICE'  , 'RUNOFFGLC'
+  glc_tavg_1hr_glo_L@HIST_IMx@HIST_JM_slv.fields:   'SNOWDP_GL'    , 'LANDICE'  , 'SNOWDPGLC'    ,
```
Comment (lcandre2) [inline HISTORY_R21C.rc.tmpl:4004]: Appended 'GLC' for consistency with LAND.

**Comment by sanAkel** [comment]
> > @elakkraoui and _others_, here is a suggested addition that writes out `2d` fields for the ocean surface.
> > [ocean_forcing.rc.txt](https://github.com/GEOS-ESM/GEOSgcm_App/files/9569850/ocean_forcing.rc.txt)
> 
> @sanAkel, here you have RUNSURF for runoff. This does not include glacier runoff. Do you want me to change this?

That would be good. I am glad you identified this fault. Thank you @lcandre2

File: HISTORY_R21C.rc.tmpl (hunk 1 above)
Comment (gmao-rreichle) [reply HISTORY_R21C.rc.tmpl:1036]: FYI, "lnd" outputs are:
- EVLAND = total evaporation in kg/m2/s
- LHLAND = total latent heat flux in W/m2
- EVPINTR, EVPSBLN, EVPSOIL, EVPTRNS = evaporation components in W/m2 (should sum up to LHLAND) 

All of the above are gridded outputs from SURFACE where the tile-to-grid aggregation includes only across "land" tiles.

Comment (sanAkel) [reply HISTORY_R21C.rc.tmpl:1036]: To add to what @gmao-rreichle said ⬆️ when ocean collection puts output, land is _masked_, same applies for land outputs: ocean masked.

Makes sense?

Comment (lcandre2) [reply HISTORY_R21C.rc.tmpl:1036]: The concern is that in the flx collection **EVAP** is exported from SURFACE, while in the ocn and glc collections **EVAPOUT** is exported from SURFACE and LANDICE, respectively. EVAP and EVAPOUT from surface do not produce  the same fields (see attached figure), suggesting an inconsistency. I looked at EVLAND and it is most similar to EVAP (not EVAPOUT). 

Rolf and I noticed the differences when going through HISTORY_R21C.rc.tmpl. I don't have any particular insight into this, and perhaps it isn't worth pursuing, but the differences are non-negligible.

@elakkraoui  @lltakacs 

evapdifferencesM11

**Comment by gmao-rreichle** [comment]
Many thanks, @sdrabenh, for merging this GEOSgcm_App PR, and especially for fixing the double comma and DRPAR & DFPAR export variable name issues during the merge. 
I double-checked that everything that was in the branch was merged ok, so I'm going to delete my branch. For any future changes that may be needed, we'll start with another clone of your most recent branch.
I'll also transfer the remaining to-do items into a new GEOSgcm_App Github issue.
cc: @elakkraoui @lcandre2

**Pull request merged**